server/src/main/java/io/crate/types/RegprocType.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.types;
+
+import io.crate.Streamer;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+import static io.crate.metadata.pgcatalog.OidHash.functionOid;
+
+public class RegprocType extends DataType<Regproc> implements Streamer<Regproc> {
+
+    public static final RegprocType INSTANCE = new RegprocType();
+    public static final int ID = 20;
+
+    @Override
+    public int id() {
+        return ID;
+    }
+
+    @Override
+    public Precedence precedence() {
+        return Precedence.REGPROC;
+    }
+
+    @Override
+    public String getName() {
+        return "regproc";
+    }
+
+    @Override
+    public Streamer<Regproc> streamer() {
+        return this;
+    }
+
+    @Override
+    public Regproc implicitCast(Object value) throws IllegalArgumentException, ClassCastException {
+        if (value == null) {
+            return null;
+        }
+        if (value instanceof Integer) {
+            return Regproc.of((int) value, value.toString());
+        } else if (value instanceof String) {
+            var name = (String) value;
+            return Regproc.of(functionOid(name), name);
```

Comment:
`'<text>'::regproc` in PostgreSQL the cast function will try to look up for the function that corresponds to the string literal. For now, I allowed this cast but it just creates a `Regproc` object with oid and name, should we forbid the text to `regproc` cast or keep it?